griptape/engines/rag/rag_engine.py

```diff
@@ -0,0 +1,26 @@
+from typing import Optional
+from attrs import define, field
+from griptape.engines.rag import RagContext
+from griptape.engines.rag.stages import QueryStage, GenerationStage, RetrievalStage
+
+
+@define(kw_only=True)
+class RagEngine:
```

Comment:
Yeah, it can be a "query engine" by having an empty generation stage. May be we can think of a way to let users pick better defaults for different RAG and more generic querying scenarios. Would rather have it be a different PR though.